### ISSUE TITLE
I found a vulnerability that could potentially be exploited by an attacker. The vulnerability lies in the setPlatformFee() function, which is used to set the platform fee.

### ISSUE THREAD

**Issue by c4-bot-7** [code-423n4/redacted-bug-bounty#61]
# Lines of code

https://github.com/redacted-cartel/pirex-eth-contracts/blob/11f30c7e35b67d45deefe405c22a30f352bc5b21/src/AutoPxEth.sol#L220

# Vulnerability details

 I found a vulnerability that could potentially be exploited by an attacker. The vulnerability lies in the `setPlatformFee()` function, which is used to set the platform fee.

**Issue:**
The `setPlatformFee()` function is accessible only by the contract owner, which is a good practice. However, the function does not include any checks to prevent the platform fee from being set to a value of 0. This can be potentially exploited by an attacker who gains control of the contract ownership, as a platform fee of 0 would result in no fees being charged to users, leading to a loss of revenue for the platform.

**Impact:**
The impact of this vulnerability is significant, as it can lead to a loss of revenue for the platform. If an attacker gains control of the contract ownership, they can set the platform fee to 0, allowing users to transact without any fees. This can result in a significant loss of revenue for the platform, especially if the attacker is able to maintain control for a prolonged period of time.

**Recommendation:**
To mitigate this vulnerability, I recommend adding a check in the `setPlatformFee()` function to prevent the platform fee from being set to a value of 0. This can be achieved by adding the following line of code:

`require(fee > 0, "Platform fee cannot be set to 0.");`

This will ensure that the platform fee cannot be set to a value of 0, preventing the loss of revenue for the platform.

**Disclosure:**
I recommend disclosing this vulnerability to the contract owner as soon as possible, so that they can take appropriate action to mitigate the risk. It is important to ensure that the vulnerability is addressed before it can be exploited by an attacker.

**Conclusion:**
In conclusion, the Smart Contract provided has a vulnerability in the `setPlatformFee()` function that can potentially be exploited by an attacker to gain control of the contract ownership and set the platform fee to 0, leading to a loss of revenue for the platform. I recommend adding a check in the `setPlatformFee()` function to prevent the platform fee from being set to a value of 0, and disclosing the vulnerability to the contract owner as soon as possible.

I hope this report helps in securing the Smart Contract and preventing any potential attacks.
